src/test/py/bazel/bzlmod/bazel_module_test.py

```diff
@@ -910,6 +910,116 @@ def testInclude(self):
     _, stdout, _ = self.RunBazel(['run', '//:main'])
     self.assertIn('main function => aaa@1.0', stdout)
 
+  def testNonRegistryOverrideModuleInclude(self):
+    """Tests that include() works in non-root modules with a non-registry override."""
+    self.ScratchFile(
+        'MODULE.bazel',
+        [
+            'module(name="aaa", version="0.1")',
+            'bazel_dep(name="bbb", version="1.0")',
+            'local_path_override(module_name="bbb", path="code_for_b")',
+        ],
+    )
+    self.AddBazelDep('rules_shell')
+    self.ScratchFile(
+        'code_for_b/MODULE.bazel',
+        [
+            'module(name="bbb", version="1.0")',
+            'include("//subdir:bbb.MODULE.bazel")',
+        ],
+    )
+    self.AddBazelDep('rules_shell', path='code_for_b')
+    self.ScratchFile('code_for_b/subdir/BUILD')
+    self.ScratchFile(
+        'code_for_b/subdir/bbb.MODULE.bazel',
+        [
+            'include("//subdir/nested:ccc.MODULE.bazel")',
+        ],
+    )
+    self.ScratchFile('code_for_b/subdir/nested/BUILD')
+    self.ScratchFile(
+        'code_for_b/subdir/nested/ccc.MODULE.bazel',
+        [
+            'bazel_dep(name="ccc", version="2.0")',
```

Comment:
What happens when the MODULE.bazel-like file included in the non-registry overridden module contain overrides of other modules? And what if there are conflicts among multiple of them?

Reply:
Oh, I assume those will be ignored because they are not rood modules anyway.